newsfragments/2981.removed.md

```diff
@@ -0,0 +1 @@
+Remove all functionality deprecated in PyO3 0.17.
```

Comment:
This should mention `acquire_gil` by name, its removal should hit a lot of people upgrading from older pyo3 versions and skipping versions.

Reply:
Done in the rebase.